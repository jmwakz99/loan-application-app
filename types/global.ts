import { Ionicons } from "@expo/vector-icons";
import { KeyboardType } from "react-native";

export type Styles = { [key: string]: string | number };

export type Size = "small" | "large";

export type ButtonVariant = "solid" | "outlined";

export type IconName = keyof typeof Ionicons.glyphMap;
export type InputField = {
  label: string;
  placeholder: string;
  type: KeyboardType;
  name: string;
};
