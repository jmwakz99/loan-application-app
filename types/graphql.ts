import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
const defaultOptions = {} as const;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
};

export type LoanApplicationObject = {
  __typename?: 'LoanApplicationObject';
  email?: Maybe<Scalars['String']['output']>;
  fullName?: Maybe<Scalars['String']['output']>;
  id?: Maybe<Scalars['Int']['output']>;
  loanAmount?: Maybe<Scalars['Float']['output']>;
  loanPurpose?: Maybe<Scalars['String']['output']>;
};

export type LoanProduct = {
  __typename?: 'LoanProduct';
  id?: Maybe<Scalars['Int']['output']>;
  interestRate?: Maybe<Scalars['Float']['output']>;
  maximumAmount?: Maybe<Scalars['Int']['output']>;
  name?: Maybe<Scalars['String']['output']>;
};

export type Query = {
  __typename?: 'Query';
  loanApplications?: Maybe<Array<Maybe<LoanApplicationObject>>>;
  loanProducts?: Maybe<Array<Maybe<LoanProduct>>>;
};

export type LoanQueryVariables = Exact<{ [key: string]: never; }>;


export type LoanQuery = { __typename?: 'Query', loanProducts?: Array<{ __typename?: 'LoanProduct', id?: number | null, name?: string | null, interestRate?: number | null, maximumAmount?: number | null } | null> | null };


export const LoanDocument = gql`
    query Loan {
  loanProducts {
    id
    name
    interestRate
    maximumAmount
  }
}
    `;

/**
 * __useLoanQuery__
 *
 * To run a query within a React component, call `useLoanQuery` and pass it any options that fit your needs.
 * When your component renders, `useLoanQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useLoanQuery({
 *   variables: {
 *   },
 * });
 */
export function useLoanQuery(baseOptions?: Apollo.QueryHookOptions<LoanQuery, LoanQueryVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<LoanQuery, LoanQueryVariables>(LoanDocument, options);
      }
export function useLoanLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<LoanQuery, LoanQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<LoanQuery, LoanQueryVariables>(LoanDocument, options);
        }
export function useLoanSuspenseQuery(baseOptions?: Apollo.SuspenseQueryHookOptions<LoanQuery, LoanQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useSuspenseQuery<LoanQuery, LoanQueryVariables>(LoanDocument, options);
        }
export type LoanQueryHookResult = ReturnType<typeof useLoanQuery>;
export type LoanLazyQueryHookResult = ReturnType<typeof useLoanLazyQuery>;
export type LoanSuspenseQueryHookResult = ReturnType<typeof useLoanSuspenseQuery>;
export type LoanQueryResult = Apollo.QueryResult<LoanQuery, LoanQueryVariables>;