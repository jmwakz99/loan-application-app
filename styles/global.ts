import { StyleSheet } from "react-native";

import { Styles } from "@/types/global";

const pressed: Styles = {
  opacity: 0.5,
};

export const globalStyles = StyleSheet.create({
  pressed,
});
