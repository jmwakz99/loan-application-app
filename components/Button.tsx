import { FC } from "react";
import { ActivityIndicator, Pressable, Text, View } from "react-native";
import Ionicons from "@expo/vector-icons/Ionicons";

import { ButtonVariant, IconName, Size } from "@/types/global";
import { COLORS } from "@/constants/colors";
import { styles } from "@/styles/button";
import { globalStyles } from "@/styles/global";
import { actuatedNormalize } from "@/utils/responsiveUtils";

type Props = {
  label: string;
  iconName?: keyof typeof Ionicons.glyphMap;
  size?: Size;
  variant?: ButtonVariant;
  loading?: boolean;
  disabled?: boolean;
  onPress?: () => void;
};

const Button: FC<Props> = ({
  label,
  iconName = "",
  size = "large",
  variant = "solid",
  loading,
  disabled,
  onPress = () => {},
}) => {
  return (
    <View
      style={[
        styles.buttonOuterContainer,
        variant === "outlined" && styles.outlinedButton,
        size === "small" && styles.buttonSmall,
      ]}
    >
      <Pressable
        onPress={onPress}
        style={({ pressed }) => pressed && [globalStyles.pressed]}
        android_ripple={{ color: COLORS.primary400 }}
      >
        <View
          style={[
            styles.buttonInnerContainer,
            (loading || disabled) && globalStyles.pressed,
            variant === "outlined" && styles.outlinedButton,
            size === "small" && styles.buttonSmall,
          ]}
        >
          {loading && <ActivityIndicator color={COLORS.light} size="small" />}
          <Text
            style={[
              styles.buttonText,
              variant === "outlined" && styles.outlinedButtonText,
              size === "small" && styles.buttonSmallText,
            ]}
          >
            {label}
          </Text>
          {iconName && (
            <Ionicons
              name={iconName as IconName}
              color={variant === "solid" ? COLORS.light : COLORS.primary700}
              size={
                size === "large" ? actuatedNormalize(24) : actuatedNormalize(12)
              }
              style={styles.buttonIcon}
            />
          )}
        </View>
      </Pressable>
    </View>
  );
};

export default Button;
