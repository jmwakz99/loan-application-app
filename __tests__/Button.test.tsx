import { fireEvent, render } from "@testing-library/react-native";

import Button from "@/components/Button";
import { ButtonVariant, IconName, Size } from "@/types/global";

describe("Button", () => {
  it("it renders correctly", () => {
    const onPressMock = jest.fn();

    const props = {
      label: "Test label",
      iconName: "wifi" as IconName,
      size: "large" as Size,
      variant: "solid" as ButtonVariant,
      loading: false,
      disabled: false,
      onPress: onPressMock,
    };

    const { getByText } = render(<Button {...props} />);

    const textLabel = getByText("Test label");

    expect(textLabel).toBeTruthy();
  });

  it("should call onPress when button is pressed", () => {
    const onPressMock = jest.fn();

    const props = {
      label: "Test label",
      iconName: "wifi" as IconName,
      size: "large" as Size,
      variant: "solid" as ButtonVariant,
      loading: false,
      disabled: false,
      onPress: onPressMock,
    };

    const { getByText } = render(<Button {...props} />);

    fireEvent.press(getByText("Test label"));

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it("snapshot", () => {
    const onPressMock = jest.fn();

    const props = {
      label: "Test label",
      iconName: "wifi" as IconName,
      size: "large" as Size,
      variant: "solid" as ButtonVariant,
      loading: false,
      disabled: false,
      onPress: onPressMock,
    };

    const button = render(<Button {...props} />).toJSON();

    expect(button).toMatchSnapshot();
  });
});
