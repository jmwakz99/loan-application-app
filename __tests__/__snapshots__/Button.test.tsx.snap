// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#30C2E3",
        "borderRadius": 40,
        "height": 92,
        "justifyContent": "center",
        "paddingHorizontal": 8,
        "width": 666,
      },
      false,
      false,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={false}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#30C2E3",
            "borderColor": "#30C2E3",
            "borderRadius": 40,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 92,
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": 666,
          },
          false,
          false,
          false,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFF",
              "fontFamily": "Inter-Bold",
              "fontSize": 32,
              "fontWeight": "600",
              "textAlign": "center",
            },
            false,
            false,
          ]
        }
      >
        Test label
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#FFFF",
              "fontSize": 48,
            },
            {
              "paddingLeft": 14,
            },
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î¼²
      </Text>
    </View>
  </View>
</View>
`;
